fabtests/common/hmem_neuron.c

```diff
@@ -325,6 +332,49 @@ int ft_neuron_memcpy_from_hmem(uint64_t device, void *dst, const void *src,
 	return 0;
 }
 
+/**
+ * @brief Get the dma-buf fd and offset for neuron memory region
+ *
+ * @param addr[in] the device buffer address
+ * @param size[in] the device buffer size (in bytes)
+ * @param fd[out] the dma-buf fd
+ * @param offset[out] the offset within the dma-buf object
+ * @return int On success, return 0. On failure, return a negative error code
+ */
+int ft_neuron_get_dmabuf_fd(void *addr, size_t size, int *fd,
+			 uint64_t *offset)
+{
+	int ret = NRT_SUCCESS;
+	struct neuron_allocation *region;
+
+	/*
+	 * The assumption is that nrt_get_dmabuf_fd() would fail for
+	 * any addr that is not the starting address of the dma-buf
+	 * object. Otherwise we need a low level op to get the base
+	 * address of the dma-buf object.
+	 */
+	*offset = ft_neuron_find_region(addr, &region);
+
+	if (neuron_ops.nrt_get_dmabuf_fd) {
+		ret = neuron_ops.nrt_get_dmabuf_fd((uintptr_t)region->ptr, size, fd);
+		if (ret != NRT_SUCCESS) {
+			FT_WARN("failed to get dmabuf fd\n");
+			return -FI_EIO;
+		}
+	} else {
+		FT_ERR("nrt_get_dmabuf_fd not found\n");
+		return -FI_EIO;
+	}
+
+	return FI_SUCCESS;
+}
+
+int ft_neuron_put_dmabuf_fd(int fd)
+{
+	close(fd);
+	return FI_SUCCESS;
+}
+
 #else
 
```

Comment:
You need to also implement these two functions as stubs for non-neuron ifdef

Reply:
fixed, thanks